api/v1alpha1/clustertemplateinstance_utils.go

```diff
@@ -292,3 +293,100 @@ func (i *ClusterTemplateInstance) GetHelmParameters(
 
 	return params, nil
 }
+
+func (i *ClusterTemplateInstance) GetSubjectsWithClusterTemplateUserRole(ctx context.Context, k8sClient client.Client) (subjects []rbacv1.Subject, err error) {
+	allRoleBindingsInNamespace := &rbacv1.RoleBindingList{}
+
+	if err := k8sClient.List(ctx, allRoleBindingsInNamespace, &client.ListOptions{
+		Namespace: i.Namespace,
+	}); err != nil {
+		return nil, err
+	}
+
+	result := []rbacv1.Subject{}
+	keys := make(map[string]bool)
+	for _, rb := range allRoleBindingsInNamespace.Items {
+		if rb.RoleRef.Kind == "ClusterRole" && rb.RoleRef.Name == "cluster-templates-user" {
+			for _, subject := range rb.Subjects {
+				key := subject.Kind + "*" + subject.Name
+				if _, value := keys[key]; !value {
+					keys[key] = true
+					result = append(result, subject)
+				}
+			}
+		}
+	}
+
+	return result, nil
+}
+
+func (i *ClusterTemplateInstance) CreateDynamicRole(ctx context.Context, k8sClient client.Client, secretNames []string) (r *rbacv1.Role, err error) {
```

Comment:
No need to pass secretNames, you can do `i.GetKubeadminPassRef()` and `i.GetKubeconfigRef()` in the function body
```suggestion
func (i *ClusterTemplateInstance) CreateDynamicRole(ctx context.Context, k8sClient client.Client, ) (r *rbacv1.Role, err error) {
```